gui/browserv7/src/RFileDialog.cxx

```diff
@@ -60,20 +60,40 @@ RFileDialog::RFileDialog(EDialogTypes kind, const std::string &title, const std:
 
    fSelect = fname;
 
-   // TODO: windows
-   fBrowsable.SetTopElement(std::make_unique<SysFileElement>("/"));
-
    auto separ = fSelect.rfind("/");
 
+   std::string workdir;
+
    if (fSelect.empty() || (separ == std::string::npos)) {
-      std::string workdir = gSystem->UnixPathName(gSystem->WorkingDirectory());
-      fBrowsable.SetWorkingDirectory(workdir);
+      workdir = gSystem->UnixPathName(gSystem->WorkingDirectory());
    } else {
-      std::string workdir = fSelect.substr(0, separ);
-      fBrowsable.SetWorkingDirectory(workdir);
+      workdir = fSelect.substr(0, separ);
       fSelect = fSelect.substr(separ+1);
    }
 
+#ifdef _MSC_VER
+   // TODO: in case of windows list of letters are required
+
+   std::string toplbl, topdir;
+
+   auto pos = workdir.find(":");
+   if (pos != std::string::npos) {
+      toplbl = workdir.substr(0,pos+1);
+   } else {
+      workdir = toplbl = "c:";
+   }
+   topdir = toplbl + "\\";
```

Comment:
Maybe you can use [PathGetDriveNumber](https://docs.microsoft.com/en-us/windows/win32/api/shlwapi/nf-shlwapi-pathgetdrivenumbera?redirectedfrom=MSDN)? (just wondering...)